book/src/erasure.md

```diff
@@ -0,0 +1,29 @@
+# Raptor coding
+
+## Summary
+This design covers the usage of RaptorQ coding in the data plane to combat packet loss.
+Every transmitted coding blob in a slot will carry the three new fields in its header.
+
+* `start_index: u64` - index of the first data blob in this erasure set.
+* `num_data: u8` - number of data blobs in this erasure set
+* `num_coding: u8` - number of coding blobs in this erasure set.
+
+The core of the change is to only transmit coding blobs. All transmitted blobs will be erasure-coded, blocktree will construct the data blobs upon receipt of a sufficient number of coded blobs.
```

Comment:
raptor doesn't require that we spell out how many data and how many coding, does it?

Reply:
what if we ran raptor on every blob with a 1400 byte MTU?

Reply:
It doesn't that data added to the end of the transmitted blobs is just there to enable blocktree to reconstruct things properly. Actually now I think that's not necessary at all.

If we did that I think we'd just rely on the signature of the reconstructed data?

Reply:
Or sign each of those encoded packets.

Reply:
we can't use signature of reconstructed data because we have to store all the data that hasn't yet re-congealed into a blob, which constitutes a spam vulnerability

signing each encoded packet should be understood performance-wise.  how long does it take us to encrypt a hash?  what if we had to encrypt, on average, 1.5k more hashes per blob?